lib/galaxy/jobs/runners/slurm.py

```diff
@@ -142,8 +158,11 @@ def __check_memory_limit( self, efile_path ):
                     f.seek(-2048, os.SEEK_END)
                     f.readline()
                 for line in f.readlines():
-                    if line.strip() == SLURM_MEMORY_LIMIT_EXCEEDED_MSG:
-                        return True
+                    stripped_line = line.strip()
+                    if stripped_line == SLURM_MEMORY_LIMIT_EXCEEDED_MSG:
+                        return 'This job was terminated because it used more memory than it was allocated.'
+                    elif any(_ in stripped_line for _ in SLURM_MEMORY_LIMIT_EXCEEDED_PARTIAL_WARNINGS):
+                        return 'This job was cancelled probably because it used more memory than it was allocated.'
```

Comment:
No, this function is called only when `slurm_state` is already `'CANCELLED'`, it will only change the `fail_message`.